src/WordSearch/WordSearchSolver.ts

```diff
@@ -0,0 +1,98 @@
+import trie = require('trie-prefix-tree');
+import {Point} from './Point';
+import {WordSearchResult} from './WordSearchResult';
+
+export class WordSearchSolver {
+  // The grid of letters which makes up the wordsearch puzzle
+  private _matrix: string[][];
+  // Transpositions to apply when finding words
+  private _directions: number[][];
+  // Potential words to find in the letter grid
+  private _targets: any;  // tslint:disable-line
+
+  constructor(matrix: string[][]) {
+    this._matrix = matrix;
+
+    this._directions = [
+      // Cardinal directions
+      [0, 1], [-1, 0], [1, 0], [0, -1],
+      // Diagonal
+      [1, 1], [-1, -1], [1, -1], [-1, 1]
+    ];
+
+    this._targets = trie([]);
+  }
+
+  findWords(words: string[]): WordSearchResult[] {
+    for (const word of words) {
+      if (word !== '') {
+        this._targets.addWord(word.trim());
+      }
+    }
+    return this.search();
+  }
+
+  private search(): WordSearchResult[] {
+    let results: WordSearchResult[] = [];
+    for (let y = 0; y < this._matrix.length; y++) {

```

Comment:
nit: At least at one point it was better practice to pull the length out to a local before the loop to avoid unnecessary reads on each iteration.